lib/src/embed_builder.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:nyxx/nyxx.dart';
+
+extension EmbedExtension on Embed {
+  EmbedBuilder toEmbedBuilder() {
+    return EmbedBuilder(
+      author: author._toEmbedAuthorBuilder(),
+      color: color,
+      description: description,
+      fields: fields?.map((field) => field._toEmbedFieldBuilder()).toList(),
+      footer: footer._toEmbedFooterBuilder(),
```

Comment:
The extension could be changed from on `EmbedFooter?` to `EmbedFooter` and accessed conditionally: `footer?._toEmbed...` - cleaning up the code in the extension itself.

Also applies to the other extensions.